### PR TITLE
COMPAT: avoid passing copy keyword to concat and astype for pandas 3.0

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
This fixes some warnings in the dev build:

```
geopandas/tests/test_geodataframe.py: 54 warnings
geopandas/tests/test_pandas_methods.py: 8 warnings
geopandas/tools/tests/test_sjoin.py: 250 warnings
  /home/runner/work/geopandas/geopandas/geopandas/tools/sjoin.py:425: DeprecationWarning: The copy keyword is deprecated and will be removed in a future version. Copy-on-Write is active in pandas since 3.0 which utilizes a lazy copy mechanism that defers copies until necessary. Use .copy() to make an eager copy if necessary.
    joined = pd.concat([left, right], axis=1, copy=False)

geopandas/tests/test_overlay.py: 12 warnings
  /home/runner/work/geopandas/geopandas/geopandas/tools/overlay.py:19: DeprecationWarning: The copy keyword is deprecated and will be removed in a future version. Copy-on-Write is active in pandas since 3.0 which utilizes a lazy copy mechanism that defers copies until necessary. Use .copy() to make an eager copy if necessary.
    df.rename(
```

### CODE REVIEW

File: geopandas/tools/overlay.py [hunk 1]
```diff
@@ -14,12 +15,19 @@ def _ensure_geometry_column(df):
     If another column with that name exists, it will be dropped.
     """
     if not df._geometry_column_name == "geometry":
```
Comment (jorisvandenbossche) [inline geopandas/tools/overlay.py:17]: (We might want to check why the code for symm diff actually requires this and couldn't be updated to be more robust regarding geometry column name, though)

File: geopandas/tools/overlay.py [hunk 2]
```diff
@@ -14,12 +15,19 @@ def _ensure_geometry_column(df):
...
-            df.drop("geometry", axis=1, inplace=True)
-        df.rename(
-            columns={df._geometry_column_name: "geometry"}, copy=False, inplace=True
-        )
-        df.set_geometry("geometry", inplace=True)
+        if PANDAS_GE_30:
+            if "geometry" in df.columns:
+                df = df.drop("geometry", axis=1)
+            df = df.rename(columns={df._geometry_column_name: "geometry"})
+            df = df.set_geometry("geometry")
```
Comment (jorisvandenbossche) [inline geopandas/tools/overlay.py:22]: This actually looks like a `rename_geometry`

**Comment by m-richards** [comment]
This is overlapping a lot with #3236, you might want to grab a version of the astype changes from there as well?

**Comment by jorisvandenbossche** [comment]
@m-richards sorry, forgot about your PR! Will indeed add the relevant bits of your PR here, and then can keep your PR to suppress the alignment warnings (which don't need to be backported)

**Review by martinfleis — APPROVED** [review]

**Pull request merged**